File: 1071-greatest-common-divisor-of-strings/1071-greatest-common-divisor-of-strings.py
class Solution:
    def gcdOfStrings(self, str1: str, str2: str) -> str:
        len1, len2 = len(str1), len(str2)
        while len2:
            len1, len2 = len2, len1 % len2
        gcd_length = len1
        answer = str1[:gcd_length]
        if answer * (len(str1) // gcd_length) == str1 and answer * (len(str2) // gcd_length) == str2:
            return answer
        return ""

File: 1071-greatest-common-divisor-of-strings/1071-greatest-common-divisor-of-strings.go
func gcdOfStrings(str1 string, str2 string) string {                   
    len1, len2 := len(str1), len(str2)
    for len2 != 0 {
        len1, len2 = len2, len1 % len2
    }
    gcdLength := len1
    answer := str1[:gcdLength]
    if strings.Repeat(answer, len(str1)/gcdLength) == str1 && strings.Repeat(answer, len(str2)/gcdLength) == str2 {
        return answer
     }
    return ""
}